File: Java/opentraceability/src/main/java/opentraceability/queries/EPCISQueryResults.java
package opentraceability.queries;

import java.util.ArrayList;
import java.util.List;

import opentraceability.models.events.EPCISQueryDocument;

public class EPCISQueryResults {

    public EPCISQueryDocument Document = null;

    public List<EPCISQueryStackTraceItem> StackTrace = new ArrayList<>();

    public List<EPCISQueryError> Errors = new ArrayList<>();

    public void merge(EPCISQueryResults results) {
        for (EPCISQueryStackTraceItem el : results.StackTrace) {
            this.StackTrace.add(el);
        }

        for (EPCISQueryError el : results.Errors) {
            this.Errors.add(el);
        }

        if (this.Document == null) {
            this.Document = results.Document;
        } else if (results.Document != null) {
            this.Document.merge(results.Document);
        }
    }
}

File: Java/OpenTraceability/src/main/opentraceability/queries/epcisqueryresults.kt
package opentraceability.queries

import java.util.*
import opentraceability.models.identifiers.*
import opentraceability.models.events.*
import opentraceability.models.events.EPCISQueryDocument
import org.intellij.markdown.lexer.push

class EPCISQueryResults {

    var Document: EPCISQueryDocument? = null

    var StackTrace: ArrayList<EPCISQueryStackTraceItem> = ArrayList<EPCISQueryStackTraceItem>()

    var Errors: ArrayList<EPCISQueryError> = ArrayList<EPCISQueryError>()

    fun Merge(results: EPCISQueryResults) {

        results.StackTrace.forEach { el->
            this.StackTrace.push(el)
        }

        results.Errors.forEach { el->
            this.Errors.push(el)
        }

        if (this.Document == null)
        {
            this.Document = results.Document;
        }
        else if (results.Document != null)
        {
            this.Document!!.Merge(results.Document!!);
        }
    }
}
